catalog/spatial/registry/registry-publication-action-provider/src/main/java/org/codice/ddf/registry/publication/manager/RegistryPublicationManager.java

```diff
@@ -94,53 +93,47 @@ public void handleEvent(Event event) {
         }
     }
 
-    public void init() throws FederationAdminException, PrivilegedActionException {
-
-        // Change from here
-        // Remove this subject stuff and change to use Security.runAsAdmin() once the PrivilegedException option has been implemented
-        Set<Principal> principals = new HashSet<>();
-        String localRoles = System.getProperty(KARAF_LOCAL_ROLES, "");
-        for (String role : localRoles.split(",")) {
-            principals.add(new RolePrincipal(role));
-        }
-        Subject subject = new Subject(true, principals, new HashSet(), new HashSet());
-
-        PrivilegedExceptionAction<List<Metacard>> federationAdminServiceAction =
-                () -> federationAdminService.getRegistryMetacards();
-        List<Metacard> metacards = subject.doAs(subject, federationAdminServiceAction);
-        // To here
-
-        for (Metacard metacard : metacards) {
-            Attribute registryIdAttribute =
-                    metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID);
-            if (registryIdAttribute == null || StringUtils.isBlank(registryIdAttribute.getValue()
-                    .toString())) {
-                LOGGER.warn("Warning metacard (id: {}} did not contain a registry id.",
-                        metacard.getId());
-                continue;
-            }
-            Attribute locations =
-                    metacard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
-
-            String registryId = registryIdAttribute.getValue()
-                    .toString();
-            if (locations != null) {
-                publications.put(registryId,
-                        Collections.unmodifiableList(locations.getValues()
-                                .stream()
-                                .map(Object::toString)
-                                .collect(Collectors.toList())));
-            } else {
-                publications.put(registryId, Collections.emptyList());
+    public void init() {
+
+        try {
+            List<Metacard> metacards = Security.runAsAdminWithException(() -> federationAdminService.getRegistryMetacards());
+
+            for (Metacard metacard : metacards) {
+                Attribute registryIdAttribute = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID);
+                if (registryIdAttribute == null || StringUtils
+                        .isBlank(registryIdAttribute.getValue().toString())) {
+                    LOGGER.warn("Warning metacard (id: {}} did not contain a registry id.", metacard.getId());
```

Comment:
I think warn applies because the metacards being iterated are registry metacards and should have registryIds. It's not a likely case that but if a registry metacard doesn't have registry id it should call attention to itself.